fendermint/app/src/snapshot.rs

```diff
@@ -0,0 +1,77 @@
+// Copyright 2022-2023 Protocol Labs
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+use crate::BlockHeight;
+use fendermint_vm_interpreter::fvm::state::{FvmStateParams, Snapshot};
+use fvm_ipld_blockstore::Blockstore;
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+
+/// The config params for snapshot
+#[derive(Clone)]
+pub struct SnapshotConfig {
+    path: String,
```

Comment:
```suggestion
    path: PathBuf,
```

I encourage that we store the most specific type in this config and push out validation to the time where the app config is parsed, rather than maintain the possibility that it's an invalid path all along the way.

The other remark is that it could be made clear by the name whether this is pointing at a directory or what.